### DISCUSSION TITLE
Single Core Build - Without Mutex

### DISCUSSION THREAD

**Discussion by ktimam** [jrouwe/JoltPhysics#735] [Q&A]
Is there a way to build jolt without mutex?

**Comment by jrouwe** [comment]
No there is not, but you can access the physics data without using mutexes through ```PhysicsSystem::GetBodyInterfaceNoLock()```, ```PhysicsSystem::GetBodyLockInterfaceNoLock``` and ```PhysicsSystem::GetNarrowPhaseQueryNoLock()```.

**Reply by ktimam** [reply]
Thanks for the pointers. I tried using GetBodyInterfaceNoLock in the HelloWorld sample, but it still includes std::mutex in the    build. I was hoping that there's a way not to include it!

**Reply by jrouwe** [reply]
The only way to remove std::mutex is to modify the code. If  you look at Mutex.h you can use the block in JPH_PLATFORM_BLUE as an example of how to implement the mutex class (just leave all functions empty).

Can I ask what the problem is with including std::mutex?

**Reply by ktimam** [reply]
I guess i'll have to edit the code then, thank you. 
The reason i need mutex removed is that the platform i'm building for is single threaded and doesn't have the mutex class.

**Reply by jrouwe** [reply]
Ok. Another trick could be to create a 'mutex' file that contains a dummy implementation of std::mutex and std::shared_mutex and make sure that it is in the include path.
